### DISCUSSION TITLE
useSelector

### DISCUSSION THREAD

**Discussion by lishine** [LegendApp/legend-state#271] [General]
Observer uses `useSelector` as well. So where is the performance gain coming from? Is it from using one `useSelector` per component instead of multiple ?

<img width="988" alt="image" src="https://github.com/LegendApp/legend-state/assets/6741645/0b8adaf2-c9fb-491b-989c-31e4eaef47ee">

**Comment by jmeistrich** [comment]
Yes, exactly. In our benchmarks we find the performance cost of observer to be the same as the cost of `useSelector`, which suggests there’s little to no overhead beyond its use of `useSelector`. And then each additional use of `useSelector` (or any hook) adds extra overhead. But `observer` wraps it all into one hook call, so it has a fixed cost of one hook.
